src/main/java/ch/njol/skript/expressions/EffMakeEggHatch.java

```diff
@@ -0,0 +1,79 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerEggThrowEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Make Egg Hatch")
+@Description("Makes the egg hatch in a Player Egg Throw event.")
+@Examples({
+	"on player egg throw:",
+	"\t# EGGS FOR DAYZ!",
+	"\tmake the egg hatch"
+})
+@Events("Egg Throw")
+@Since("INSERT VERSION")
+public class EffMakeEggHatch extends Effect {
+
+	static {
+		Skript.registerEffect(EffMakeEggHatch.class,
+			"make [the] egg [:not] hatch"
+		);
+	}
+
+	private boolean not;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(PlayerEggThrowEvent.class)) {
+			Skript.error("You can't use the 'make the egg hatch' effect outside of a Player Egg Throw event.");
+			return false;
+		}
+		not = parseResult.hasTag("not");
+		return true;
+	}
+
+	@Override
+	protected void execute(Event e) {
+		if (e instanceof PlayerEggThrowEvent) {
```

Comment:
I don't think we need to be checking the event instance. This happens in all classes, there is only 1 event and will always be the single event, and we're not going to change that as this event is secluded.

Reply:
> We need to be checking the event instance.

Great, because that's what is done here. 
If you meant we *don't* need to check it: we do, see #4884. TLDR isCurrentEvent checks only one of the currentEvents, not all. The fact that `on egg throw` is only registered with 1 Event doesn't matter, because an addon can still register for example `on throw` with multiple Event classes, in which case using this syntax will throw an CCE at runtime for the other Events it was registered with.

Reply:
Well that's not Skript's fault at that point, and having this performance check each execute is bad for an "assumed" issue.

Maybe we need a new method inside the SyntaxElement that returns a boolean depending on the exact event that it will call, rather than "oh it may be this event" or just return false in the init if the event wasn't registered by Skript, as Skript already has the event handled itself.

Reply:
This was recently changed in https://github.com/SkriptLang/Skript/pull/4884. If you think this isn't the solution, feel free to open a discussion suggesting a better solution. Regardless, it is not specific to this PR.